x-pack/plugins/index_lifecycle_management/public/store/selectors/lifecycle.js

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  PHASE_HOT,
+  PHASE_WARM,
+  PHASE_COLD,
+  PHASE_DELETE,
+  PHASE_ENABLED,
+  PHASE_ROLLOVER_ENABLED,
+  PHASE_ROLLOVER_MAX_AGE,
+  PHASE_ROLLOVER_MAX_SIZE_STORED,
+  STRUCTURE_INDEX_TEMPLATE,
+  STRUCTURE_CONFIGURATION,
+  STRUCTURE_PRIMARY_NODES,
+  STRUCTURE_REPLICAS,
+  STRUCTURE_TEMPLATE_SELECTION,
+  STRUCTURE_TEMPLATE_NAME,
+  STRUCTURE_POLICY_NAME,
+  STRUCTURE_POLICY_CONFIGURATION,
+  STRUCTURE_INDEX_NAME,
+  STRUCTURE_ALIAS_NAME,
+  ERROR_STRUCTURE,
+  PHASE_ATTRIBUTES_THAT_ARE_NUMBERS,
+  PHASE_PRIMARY_SHARD_COUNT,
+  PHASE_SHRINK_ENABLED,
+  STRUCTURE_REVIEW,
+  PHASE_FORCE_MERGE_ENABLED,
+  PHASE_FORCE_MERGE_SEGMENTS
+} from '../constants';
+import {
+  getPhase,
+  getPhases,
+  phaseToES,
+  getSelectedPolicyName,
+  getSelectedIndexTemplateName,
+  isNumber,
+  getSelectedPrimaryShardCount,
+  getSelectedReplicaCount,
+  getSaveAsNewPolicy,
+  getSelectedOriginalPolicyName,
+  getBootstrapEnabled,
+  getIndexName,
+  getAliasName,
+} from '.';
+
+export const validatePhase = (type, phase) => {
+  const errors = {};
+
+  if (!phase[PHASE_ENABLED]) {
+    return errors;
+  }
+
+  if (phase[PHASE_ROLLOVER_ENABLED]) {
+    if (
+      !isNumber(phase[PHASE_ROLLOVER_MAX_AGE]) &&
+      !isNumber(phase[PHASE_ROLLOVER_MAX_SIZE_STORED])
+    ) {
+      errors[PHASE_ROLLOVER_MAX_AGE] = [
+        'A maximum age is required'
+      ];
+      errors[PHASE_ROLLOVER_MAX_SIZE_STORED] = [
+        'A maximum index size is required'
+      ];
+    }
+  }
+
+  for (const numberedAttribute of PHASE_ATTRIBUTES_THAT_ARE_NUMBERS) {
+    if (phase.hasOwnProperty(numberedAttribute) && phase[numberedAttribute] !== '') {
+      // If shrink is disabled, there is no need to validate this
+      if (numberedAttribute === PHASE_PRIMARY_SHARD_COUNT && !phase[PHASE_SHRINK_ENABLED]) {
+        continue;
+      }
+      if (!isNumber(phase[numberedAttribute])) {
```

Comment:
This might be a little easier to scan if we "early continue" in these conditions:

```js
      if (!isNumber(phase[numberedAttribute])) {
        errors[numberedAttribute] = ['A number is required'];
        continue;
      }
      if (phase[numberedAttribute] < 0) {
        errors[numberedAttribute] = ['Only positive numbers are allowed'];
        continue;
      }
      if (numberedAttribute === PHASE_PRIMARY_SHARD_COUNT && phase[numberedAttribute] < 1) {
        errors[numberedAttribute] = ['Only positive numbers are allowed'];
        continue;
      }
```